src/Utilities/ExtensionSDK.cs

```diff
@@ -143,36 +137,18 @@ public Version MaxPlatformVersion
         /// <summary>
         /// Api contracts from the SDKManifest, if any
         /// </summary>
-        public ICollection<ApiContract> ApiContracts
-        {
-            get
-            {
-                return Manifest.ApiContracts;
-            }
-        }
+        public ICollection<ApiContract> ApiContracts => Manifest.ApiContracts;
 
         /// <summary>
         /// Reference to the manifest object
         /// Makes sure manifest is instantiated only once
         /// </summary>
-        private SDKManifest Manifest
-        {
-            get
-            {
-                if (_manifest == null)
-                {
-                    // Load manifest from disk the first time it is needed
-                    _manifest = new SDKManifest(_path);
-                }
-
-                return _manifest;
-            }
-        }
+        private SDKManifest Manifest => _manifest ?? (_manifest = new SDKManifest(_path));
```

Comment:
Field initializers can't access non-statics for some reason, so we'd have to introduce a ctor to set up the Lazy. It feels like this `foo ?? (foo = CreateFoo())` is just as good a Lazy though. Any strong opinion on this?

---
In reply to: [197166597](https://github.com/Microsoft/msbuild/pull/3438#discussion_r197166597) [](ancestors = 197166597)

Reply:
It's almost as good, except it can be racy where Lazy guarantees single init. But it doesn't matter in most cases, as long as the objects are interchangeable. One thread constructs and gets one object, another constructs and gets another, subsequent calls see the latter.

I think this can stay as it is in the PR now.